### PR TITLE
Fixes #18088: Generate documentation for the stdlib

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/18088

### CODE REVIEW

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Do not merge, the new stdlib breaks the parser

**Comment by amousset** [comment]
PR updated with a new commit

File: rudder-lang/src/generators/documentation.rs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+                for parameter in state.parameters.iter() {
+                    println!("* {}", parameter.name.fragment())
+                }
+
+                /*
+                // This condition actually rejects every file that is not the input filename
+                // therefore preventing from having an output in another directory
+                // Solutions: check filename rather than path, or accept everything that is not from crate root lib
+                let file_to_create = match get_dest_file(source_file, state_name.file(), dest_file)
```
[the remark this answers is not in the document]
Comment (amousset) [reply rudder-lang/src/generators/documentation.rs:70]: because it is not done yet

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: rudder-lang/libs/std/dsc.rl [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+@class_prefix = "dsc_built_in_resource"
+@class_parameter_index = 1
+@parameter.scriptBlock.description = '''
+@Desired state for the resource
+@'''
+@parameter.scriptBlock.type = "HereString"
+@parameter.resourceName.description = '''
+@resourceName
+@'''
+@supported_target = []
```
Comment (amousset) [inline rudder-lang/libs/std/dsc.rl:58]: should be dsc

File: rudder-lang/libs/resourcelib.rl [hunk 3]
```diff
@@ -1,1053 +0,0 @@
-@format=0
```
Comment (amousset) [inline rudder-lang/libs/resourcelib.rl:1]: shouldn't we remove the lib from the repo?

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: rudder-lang/tools/generate_lib [hunk 4]
```diff
@@ -1,139 +1,539 @@
...
+    match_line = ""
+    bundle_name = "unknown"
+
+    for line in content.splitlines():
+        # line should already be unicode
+        # unicodeLine = unicode(line,"UTF-8") #line.decode('unicode-escape')
+
+        # Parse metadata tag line
+        match = re.match(
+            r"^\s*#\s*@(\w+)\s*(([a-zA-Z0-9_]+)?\s+(.*?)|.*?)\s*$",
```
Comment (gpoblon) [inline rudder-lang/tools/generate_lib:120]: named parameters (if possible) would make it easier to understand

Comment (amousset) [reply rudder-lang/tools/generate_lib:120]: i have to admit I just took the regexes from the ncf library

Comment (amousset) [reply rudder-lang/tools/generate_lib:120]: but I can try to improve it indeed.

**Comment by amousset** [comment]
OK, squash merging this PR

**Comment by amousset** [comment]
OK, squash merging this PR

**Comment by amousset** [comment]
OK, merging this PR

**Pull request merged**